Refactor Code Part 2

Christian-Oleson opened the pull request:
Keeping things small as I am getting to this when I have a few minutes. It'll keep the code review smaller for you as well, and we can prevent breaking things without a ton of unit tests.

Christian-Oleson committed:
Refactor Code Part 2

Keeping things small as I am getting to this when I have a few minutes. It'll keep the code review smaller for you as well, and we can prevent breaking things without a ton of unit tests.

Christian-Oleson committed:
Pt 2.2

zo0o0ot commented on src/prospectScraper/MockDraftPick.cs:
```diff
@@ -33,36 +32,17 @@ public MockDraftPick(string pick, string team, string name, string school, strin
+
             if (canParse)
             {
-                /* 
```
I'd prefer if this comment was kept, since this information is useful, and changes from year to year.

Christian-Oleson replied:
Reverted

zo0o0ot commented on src/prospectScraper/MockDraftPick.cs:
```diff
@@ -93,25 +73,12 @@ public static int ConvertPickToRound(string pick)
+            var canParse = int.TryParse(pick, out int intpick);
             if (canParse)
             {
-                /* 
```
This comment is also useful and should be kept.

Christian-Oleson replied:
Ah, my bad. I tend to have an issue with comments, so I can be aggressive removing them from codebases. I wonder if there is a way to make this information useful, within the console app, without making it a comment? 🤔

zo0o0ot reviewed (changes requested):
Please preserve the two comments I mentioned.  Other than that, things look good.

Christian-Oleson committed:
PR Concern

Christian-Oleson committed:
Merge remote-tracking branch 'upstream/master' into Refactor_Code_Part_2

Christian-Oleson commented:
> Please preserve the two comments I mentioned. Other than that, things look good.

Done. 😃

zo0o0ot commented:
Crap.  I'm sorry about this, but I just added a new parameter to handle #32 in #35 and it looks like I introduced some conflicts.

zo0o0ot commented:
Would you be willing to port over my changes to the appropriate new files that you've created? I ended up having to change more than I expected, since the date parsing was moved into a mock draft specific method and left inline for the big board scraping.  I ended up needing to make two separate methods, one that returned a DateTime and the other returning a string since apparently the way I create the file name for the csv files wasn't consistent between the mock draft and the big board.

Christian-Oleson committed:
Merge remote-tracking branch 'upstream/master' into Refactor_Code_Part_2

Christian-Oleson committed:
Merge-Conflicts-Oh my!

Christian-Oleson commented:
> Would you be willing to port over my changes to the appropriate new files that you've created? I ended up having to change more than I expected, since the date parsing was moved into a mock draft specific method and left inline for the big board scraping. I ended up needing to make two separate methods, one that returned a DateTime and the other returning a string since apparently the way I create the file name for the csv files wasn't consistent between the mock draft and the big board.

I think I just moved over everything needed? If not, please let me know. That was a nasty set of merge issues. Once we get everything properly refactored, we can put the unit tests in that will catch behavior changes if I introduce them to the project.

Christian-Oleson commented:
> Crap. I'm sorry about this, but I just added a new parameter to handle #32 in #35 and it looks like I introduced some conflicts.

And don't worry, these things happen. That's the reason I'm trying to keep the PRs smaller though, so we can get the changes in rather quickly, so we don't have to deal with merge issues. 😄

zo0o0ot commented on src/prospectScraper/Program.cs:
```diff
@@ -39,720 +33,27 @@ static void Main(string[] args)
                     default:
-                        Console.WriteLine("Input argument of " + context + " not recognized.  Please try running again.");
-                        RunTheBigBoards(parseDate);
+                        Console.WriteLine("Input argument of " + s + " not recognized. Please try running again.");
```
this line needs to be removed.  It references the old generic variable "s".

Christian-Oleson replied:
Woah, something did not push right in this merge resolution. Taking a look now...

Christian-Oleson replied:
but this is done

zo0o0ot commented on src/prospectScraper/ProspectScraper.cs:
```diff
@@ -0,0 +1,625 @@
+
+        public void RunTheMockDraft(bool parseDate)
+        {
+            //TODO - Implement Mock Draft
```
This comment can be deleted.

zo0o0ot commented on src/prospectScraper/Program.cs:
```diff
@@ -39,720 +33,27 @@ static void Main(string[] args)
-            return mdp;
-        }
-
-        public static DateTime ChangeDateStringToDateTime(string scrapedDate, bool parseDate)
```
This method is new. It needs to be in ProspectScraper.cs. 

It replaces the functionality of GetDraftDate() and FormatDraftDate()

Christian-Oleson replied:
Yeah, this got hosed in the merge conflict resolution. I'm adding it back.

zo0o0ot commented on src/prospectScraper/Program.cs:
```diff
@@ -39,720 +33,27 @@ static void Main(string[] args)
-                return DateTime.Now;
-            }
-        }
-        public static string FormatScrapedDate(DateTime scrapedDate)
```
I see that this method has been replaced with an inline `.ToString("yyyy-MM-dd");`  I'm cool with that.  Mostly I was trying to salvage the method that had been previously created, and it isn't really necessary.

Christian-Oleson replied:
I'll add more refactoring after we get this one merged in and fixed up that should make these classes smaller and more readable. 😄

zo0o0ot commented on src/prospectScraper/ProspectScraper.cs:
```diff
@@ -0,0 +1,625 @@
+
+            //Get ranking date
+            var dateOfRanks = document1.DocumentNode.SelectSingleNode("//*[@id='HeadlineInfo1']").InnerText.Replace(" EST", "").Trim();
+            var parsedDate = ChangeDateStringToDateTime(dateOfRanks, parseDate);
```
Here's the method, but it does not exist in ProspectScraper.cs

Christian-Oleson replied:
Fixed up. It pulled a Houdini.

zo0o0ot commented on src/prospectScraper/ProspectScraper.cs:
```diff
@@ -0,0 +1,625 @@
+            HtmlNode hn = document1.DocumentNode;
+            HtmlNode hi1 = hn.SelectSingleNode("//*[@id='HeadlineInfo1']");
+            DateTime mockDraftDate = ChangeDateStringToDateTime(hi1.InnerText.Replace(" EST", "").Trim(), parseDate);
+            string draftDate = FormatScrapedDate(mockDraftDate);
```
This method call can be replaced with 
`string draftDate = mockDraftDate.ToString("yyyy-MM-dd");`

zo0o0ot replied:
This `FormatScrapedDate` method can be replaced with:
```suggestion
            string draftDate = mockDraftDate.ToString("yyyy-MM-dd");
```
`string draftDate = mockDraftDate.ToString("yyyy-MM-dd");`

Christian-Oleson committed:
Bad push, undo

Christian-Oleson commented:
@zo0o0ot ,
You should see the new changes reflected shortly. I think I forgot to stage some of the changes, and then discarded some other changes, in the attempt to merge things up. Unit tests ran initially, but because I didn't push everything, it left this in a non-working state. Unit tests are passing again, everything is building, and fingers crossed we should be good to go here.

zo0o0ot commented on src/prospectScraper/ProspectScraper.cs:
```diff
@@ -0,0 +1,625 @@
+            return mdp;
+        }
+
+        public static string GetDraftDate(HtmlDocument doc)
```
This method is no longer needed.

zo0o0ot commented on src/prospectScraper/ProspectScraper.cs:
```diff
@@ -0,0 +1,625 @@
+            return FormatDraftDate(doc.DocumentNode.SelectSingleNode("//*[@id='HeadlineInfo1']").InnerText);
+        }
+
+        public static string FormatDraftDate(string headlineInfo)
```
This method is no longer needed.

Christian-Oleson replied:
This is referenced in the tests. I need to unfurl that at some point. We want to remove as much as possible static classes and fields, as they tightly couple the code.

zo0o0ot reviewed (changes requested):
See comments for the suggested changes, including the method changes.

Christian-Oleson committed:
PR Concerns

Christian-Oleson commented:
> See comments for the suggested changes, including the method changes.

Looks like I was about 1 minute behind you. Let me know if there is anything else we need to hit. 😄

Christian-Oleson committed:
PR Concerns

Christian-Oleson committed:
A few more modifications

That's it for tonight, and probably the weekend. If no issues arise, that is.

Christian-Oleson commented:
> > See comments for the suggested changes, including the method changes.
> 
> Looks like I was about 1 minute behind you. Let me know if there is anything else we need to hit. 😄

@zo0o0ot ,
Also went in and refactored a tad more once I cleaned that up. I think this is starting to take form. 😄

Christian-Oleson commented on src/prospectScraper/ProspectScraper.cs:
```diff
@@ -0,0 +1,636 @@
+            {
+                UserAgent = "Mozilla/5.0 (Windows NT 6.3; WOW64; rv:31.0) Gecko/20100101 Firefox/31.0"
+            };
+            var document1 = webGet.Load("https://www.drafttek.com/2021-NFL-Draft-Big-Board/Top-NFL-Draft-Prospects-2021-Page-1.asp");
+            var document2 = webGet.Load("https://www.drafttek.com/2021-NFL-Draft-Big-Board/Top-NFL-Draft-Prospects-2021-Page-2.asp");
+            var document3 = webGet.Load("https://www.drafttek.com/2021-NFL-Draft-Big-Board/Top-NFL-Draft-Prospects-2021-Page-3.asp");
+            var document4 = webGet.Load("https://www.drafttek.com/2021-NFL-Draft-Big-Board/Top-NFL-Draft-Prospects-2021-Page-4.asp");
+            var document5 = webGet.Load("https://www.drafttek.com/2021-NFL-Draft-Big-Board/Top-NFL-Draft-Prospects-2021-Page-5.asp");
```
This is the one that I will eventually get to. We can eventually use polymorphism to allow any sort of draft, and we can find a way to make this scale based on that draft type. Prefer the MLB? NHL? NBA? Well, maybe, just maybe someday, that will be possible.

zo0o0ot replied:
Check out #25 as there is some good stuff there, but I haven't merged it yet because we're still moving stuff around all over the place.

Christian-Oleson replied:
Awesome! I'll stop moving things for now so we can get some of this in then. 👍

Christian-Oleson committed:
Merge remote-tracking branch 'upstream/master' into Refactor_Code_Part_2

Christian-Oleson committed:
Fix build step

zo0o0ot reviewed (approved):
Looks good.
